google/oauth2/service_account.py

```diff
@@ -287,18 +309,9 @@ def with_always_use_jwt_access(self, always_use_jwt_access):
             google.auth.service_account.Credentials: A new credentials
                 instance.
         """
-        return self.__class__(
-            self._signer,
-            service_account_email=self._service_account_email,
-            scopes=self._scopes,
-            default_scopes=self._default_scopes,
-            token_uri=self._token_uri,
-            subject=self._subject,
-            project_id=self._project_id,
-            quota_project_id=self._quota_project_id,
-            additional_claims=self._additional_claims.copy(),
-            always_use_jwt_access=always_use_jwt_access,
-        )
+        cred = self._make_copy()
+        cred._always_use_jwt_access = always_use_jwt_access
```

Comment:
if the input value is false and original credential has TPC universe - do we want to throw? 

Otherwise could backfire for some existing code moved into TPC env.

Reply:
added logic to throw